homeassistant/components/smartthings/climate.py

```diff
@@ -462,3 +494,55 @@ def target_temperature(self):
     def temperature_unit(self):
         """Return the unit of measurement."""
         return UNIT_MAP.get(self._device.status.attributes[Attribute.temperature].unit)
+
+    def _determine_swing_modes(self) -> list[str]:
+        """Return the list of available swing modes."""
+        supported_modes = self._device.status.attributes[
+            Attribute.supported_fan_oscillation_modes
+        ][0]
+        supported_swings = [
+            FAN_OSCILLATION_TO_SWING.get(m, SWING_OFF) for m in supported_modes
+        ]
+        return supported_swings
+
+    async def async_set_swing_mode(self, swing_mode: str) -> None:
+        """Set swing mode."""
+        fan_oscillation_mode = SWING_TO_FAN_OSCILLATION[swing_mode]
+        await self._device.set_fan_oscillation_mode(fan_oscillation_mode)
+
+        # setting the fan must reset the preset mode (it deactivates the windFree function)
+        self._attr_preset_mode = ""
+
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def swing_mode(self) -> str:
+        """Return the swing setting."""
+        return FAN_OSCILLATION_TO_SWING.get(
+            self._device.status.fan_oscillation_mode, SWING_OFF
+        )
+
+    def _determine_preset_modes(self) -> list[str]:
+        """Return a list of available preset modes."""
+        modes = []
+        supported_modes = self._device.status.attributes[
+            "supportedAcOptionalMode"
+        ].value
+        if WINDFREE in supported_modes:
+            modes.append(WINDFREE)
+        return modes
+
+    async def async_set_preset_mode(self, preset_mode):
```

Comment:
Missing typing

Reply:
Can we type `__init__` above too so it's complete?

Reply:
I added typings to all methods in the class. Thanks for the feedback